### PR TITLE
Back-port BEAM changes for display data.

### PR DESCRIPTION
**Pull request by swegner**

### CODE REVIEW

**Comment by swegner** [comment]
R: @bjchambers

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/options/ProxyInvocationHandler.java [hunk 1]
```diff
@@ -16,17 +16,23 @@
 
 package com.google.cloud.dataflow.sdk.options;
 
+import com.google.api.client.repackaged.com.google.common.base.Objects;
```
Comment (bjchambers) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/options/ProxyInvocationHandler.java]: Should not depend on repackaged versions of dependencies.

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/io/WriteTest.java [hunk 2]
```diff
@@ -151,6 +155,23 @@ public void testWriteWithSessions() {
...
+      }
+    };
+    Write.Bound<String> write = Write.to(sink);
+    DisplayData displayData = DisplayData.from(write);
+
+    assertThat(displayData, hasDisplayItem("sink", sink.getClass()));
+    assertThat(displayData, includes(sink));
+  }
+
+
```
Comment (bjchambers) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/io/WriteTest.java]: Seems like a lot of blank whitespace

**Comment by bjchambers** [comment]
Are there any possibly surprising changes here beyond just cherry-picking and fixing up packaging? I made a quick scan and everything seems sensible, want to make sure there isn't anything that merits further scrutiny.

**Comment by swegner** [comment]
Nope, the process was the cherry-pick each change and fix all the imports (added imports have org.apache.beam.* prefix).  I'll take a pass to look for any other repackaged imports.

**Comment by swegner** [comment]
I didn't come across any other bad imports. Please take another look: @bjchambers

**Pull request merged**